src/job_token.rs

```diff
@@ -1,139 +1,152 @@
-use jobserver::{Acquired, Client, HelperThread};
-use std::{
-    env,
-    mem::MaybeUninit,
-    sync::{
-        mpsc::{self, Receiver, Sender},
-        Once,
-    },
-};
-
-pub(crate) struct JobToken {
-    /// The token can either be a fresh token obtained from the jobserver or - if `token` is None - an implicit token for this process.
-    /// Both are valid values to put into queue.
-    token: Option<Acquired>,
-    /// A pool to which `token` should be returned. `pool` is optional, as one might want to release a token straight away instead
-    /// of storing it back in the pool - see [`Self::forget()`] function for that.
-    pool: Option<Sender<Option<Result<Acquired, crate::Error>>>>,
-}
+use std::{mem::MaybeUninit, sync::Once};
+
+use crate::Error;
+
+#[cfg(unix)]
+#[path = "job_token/unix.rs"]
+mod sys;
+
+#[cfg(windows)]
+#[path = "job_token/windows.rs"]
+mod sys;
+
+pub(super) struct JobToken();
```

Comment:
I guess it could be an unit struct instead of an empty tuple struct, what'd you say?
```suggestion
pub(super) struct JobToken;
```

Reply:
I deliberately wrote it like that, to prevent other modules from creating `JobToken` accidentally without calling `try_acquire`.

Reply:
Ah, I see! Thanks for clearing that up, let's keep it then.